Add blog post about Sphinx and Markdown (#147)
readthedocs/blog: sphinx-markdown-2021.rst

.. post:: Dec 20, 2021
   :tags: media, talk, conference, markdown, commonmark
   :author: Juan Luis
   :location: MAD

.. meta::
   :description lang=en:
      During 2021 we have promoted the use of Markdown in Sphinx in various ways.
      In this post we show how you can start using it in your project today,
      and give some extra pointers.

Sphinx and Markdown around the world in 2021
============================================

Read the Docs has been committed to improving the accessibility

> and user experience of Sphinx since its beginnings,

Reviewer: ```suggestion
and user experience of Sphinx since the start,
```

reads a bit better.

and that includes the markup language in which the documentation is written.
After carefully listening to users
that wanted to use Markdown in Sphinx as well as reStructuredText,
:doc:`we announced recommonmark more than six years ago </adding-markdown-support>`
to help bridging the immense popularity of Markdown

> with the powerful capabilities of Sphinx.

Reviewer 2: I found this sentence confusing when I read it. At first, I thought we listen to our users in the last months, but then I realized that it's taking about 6 years ago. However, I've already clicked on the blog post and I was reading it 😄 

If possible, I'd change the order of it to mention the time at the beginning of the sentence to immediately contextualize the reader. Something like,

```
Years ago, after carefully listening to users ... we announced recommonmark
```

Besides, I think it would be good to mention in the same paragraph that it's currently deprecated, something like:

```
which is now, deprecated in favor of MyST (keep reading to find out more about how to use it / migrate)
```

Reviewer: Yea, 👍 to mentioning that it's deprecated in the same sentence.

Author: Done ✔️

> It is no surprise that Markdown is so demanded:

Reviewer: ```suggestion
It is no surprise that Markdown is in such demand:
```

thanks in large part to the huge popularity of GitHub,
`Markdown is nowadays the most widely used markup language in open-source
projects <https://passo.uno/docs-as-code-tools-open-standards/>`_,
ahead of reStructuredText, the second in the list, by an order of magnitude.

This year, thanks to the CZI Grant we received last year
and the effort of other communities,
especially `the Executable Book Project <https://executablebooks.org>`_,
we have devoted lots of resources to help consolidate Markdown
as a compelling markup language for Sphinx.

Writing Markdown in Sphinx
--------------------------

To that end, we recognized the potential of MyST as a successor of recommonmark,
and :doc:`deprecated the latter in favor of MyST-Parser </newsletter-april-2021>`.
To :doc:`start writing Markdown in Sphinx <sphinx:usage/markdown>`,

> you need to follow three steps:

Reviewer 2: Maybe we can remove the deprecation notice from here and put it only on the initial paragraph instead.

Here we could just talk about MyST without mentioning recommonmark. If we want to highlight that recommonmark is deprecated, we could have a specific section for this, tho. I think that mentioning this deprecation here is out of the scope of the section's title and adds confusion only.

Actually, do we want to mention recommonmark in this post at all? 😄 . Maybe the focus on MyST + Sphinx + conferences is 💯 without recommonmark.

Reviewer: I think we mention it at the top, but talk mostly about the future.

* Install ``MyST-Parser``: ``pip install myst-parser``

* Add ``myst_parser`` to the list of Sphinx extensions:

.. code-block:: python
   :emphasize-lines: 3

   extensions = [
       # ...other extensions
       "myst_parser",
   ]

* Start writing content in ``*.md`` files!

You can find more information in the :doc:`MyST-Parser documentation <myst-parser:index>`.

Talks and workshops at conferences
----------------------------------

Luckily the migration to MyST-Parser was trivial for most projects,
but we still wanted to spread the word about

> the best practices for creating Sphinx projects on Read the Docs using MyST Markdown.

Reviewer 2: Now, after reading this paragraph, I seems that migrating users from recommonmark to MyST was part of this job and now, having a specific section commenting that recommonmark was deprecated in favor or MyST makes more sense.

Author: Since you made the earlier comments before reading the whole post, I prefer to wait on @ericholscher to do the proofreading. Thanks anyway @humitos !
